lib/transport/ble/gatt-client.js

```diff
@@ -310,9 +307,27 @@ class GattClient extends EventEmitter {
         return this.pairingProtocol.parsePairSetupM2(
           body[GattConstants.Types['HAP-Param-Value']]
         );
-      }).then((tlv) => {
-        return this.pairingProtocol.buildPairSetupM3(tlv, pin);
-      }).then((packet) => {
+      });
```

Comment:
`characteristic` and `iid` are going to have to be returned from this function as well, since they're used in `finishPairing()`. BLE pairing is currently broken without that.

Reply:
Ok, I think I fixed that.